src/Cimpress.Extensions.Http.Caching.InMemory/IMemoryCacheExtensions.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Threading.Tasks;
+using Cimpress.Extensions.Http.Caching.Abstractions;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace Cimpress.Extensions.Http.Caching.InMemory
+{
+    /// <summary>
+    /// Extension methods for an <see cref="IMemoryCache"/>.
+    /// </summary>
+    internal static class IMemoryCacheExtensions
+    {
+        /// <summary>
+        /// Tries to get the data from cache, that is, ignoring all exceptions.
+        /// </summary>
+        /// <param name="cache">The in memory cache.</param>
+        /// <param name="key">The key to retrieve from the cache.</param>
+        /// <returns>The data of the cache entry, or null if not found or on any error.</returns>
+        public static async Task<CacheData> TryGetAsync(this IMemoryCache cache, string key)
+        {
+            try
+            {
+                byte[] binaryData = null;
+                if (cache.TryGetValue(key, out binaryData))
+                {
+                    await Task.FromResult(true);
```

Comment:
I think this should be `return Task.FromResult(binaryData.Deserialize())` and not make this method async. This is just compiler magic overhead that's added.